apps/upgrade-journey/run.go

```diff
@@ -0,0 +1,486 @@
+
+	}
+
+	return nil
+}
+
+func createSchema(ctx context.Context, client *weaviate.Client) error {
+	refTarget := &models.Class{
+		Class: "RefTarget",
+		Properties: []*models.Property{
+			{
+				DataType: []string{"string"},
+				Name:     "version",
+			},
+			{
+				DataType: []string{"int"},
+				Name:     "object_count",
+			},
+		},
+	}
+
+	err := client.Schema().ClassCreator().WithClass(refTarget).Do(context.Background())
+	if err != nil {
+		return err
+	}
+
+	classObj := &models.Class{
+		Class: "Collection",
+		Properties: []*models.Property{
+			{
+				DataType: []string{"string"},
+				Name:     "version",
+			},
+			{
+				DataType: []string{"int"},
+				Name:     "object_count",
+			},
+			{
+				DataType: []string{"RefTarget"},
+				Name:     "ref_prop",
+			},
+			{
+				DataType: []string{"int"},
+				Name:     "major_version",
+			},
+			{
+				DataType: []string{"int"},
+				Name:     "minor_version",
+			},
+			{
+				DataType: []string{"int"},
+				Name:     "patch_version",
+			},
+		},
+	}
+
+	err = client.Schema().ClassCreator().WithClass(classObj).Do(context.Background())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func importForVersion(ctx context.Context, client *weaviate.Client,
+	version string,
+) error {
+	targetID := uuid.New().String()
+	if err := importTargetObject(ctx, client, version, targetID); err != nil {
+		return fmt.Errorf("source object: %w", err)
+	}
+
+	if err := importSourceObject(ctx, client, version, targetID); err != nil {
+		return fmt.Errorf("source object: %w", err)
+	}
+
+	objectsCreated++
+
+	return nil
+}
+
+func importTargetObject(ctx context.Context, client *weaviate.Client,
+	version, id string,
+) error {
+	props := map[string]interface{}{
+		"version":      version,
+		"object_count": objectsCreated,
+	}
+
+	_, err := client.Data().Creator().
+		WithClassName("RefTarget").
+		WithID(id).
+		WithProperties(props).
+		Do(context.Background())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func importSourceObject(ctx context.Context, client *weaviate.Client,
+	version, targetID string,
+) error {
+	semver, _ := maybeParseSingleSemverWithoutLeadingV(version)
+	props := map[string]interface{}{
+		"version":       version,
+		"object_count":  objectsCreated,
+		"ref_prop":      []interface{}{map[string]interface{}{"beacon": fmt.Sprintf("weaviate://localhost/RefTarget/%s", targetID)}},
+		"major_version": semver.major,
+		"minor_version": semver.minor,
+		"patch_version": semver.patch,
+	}
+
+	vec := make([]float32, 32)
+	for i := range vec {
+		vec[i] = rand.Float32()
+	}
+
+	_, err := client.Data().Creator().
+		WithClassName("Collection").
+		WithVector(vec).
+		WithProperties(props).
+		Do(context.Background())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func startOrUpgrade(ctx context.Context, c *cluster, i int, version string) error {
+	if i == 0 {
+		return c.startAllNodes(ctx, version)
+	}
+
+	return c.rollingUpdate(ctx, versions[i%len(versions)])
```

Comment:
why are we not using `version` instead of `versions[i%len(versions)]?

Reply:
Haha, that would be too easy, I'll change it. I think this is a leftover from some experiment I did initally.